src/test/ac-mock.ts

```diff
@@ -53,11 +56,18 @@ export class AcMock {
         this.router.post('/~vault/pan', ctx => this.createPanToken(ctx));
         this.app.use(async (ctx, next) => {
             try {
+                if (this.serverError) {
```

Comment:
This should probably take place in `src/main/job.ts:387`, which is `Job.track()`.

Reply:
The `AcMock` is a fake http server which the client talks to, so I'm convinced it's a proper place for `serverError` simulation.

One thing we can improve is to throw finer grained errors in tests (e.g. specifically when one of the endpoints is invoked) — this can be done by placing more `emit`s and throwing from them. Feel free to amend this bit, or to reproduce in any other way as you see fit — point is, we need to make this reproducible.